packages/ios_platform_images/ios/Classes/UIImageIosPlatformImages.swift

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import Foundation
+import UIKit
+
+extension UIImage {
```

Comment:
Extending framework classes is more potential pain than it is worth; we should take the opportunity of this rewrite to make this a helper function rather than a static class extension.

Reply:
Sorry, I forgot that this is one of the rare cases of a method intended for use outside of the plugin, so changing its structure would be a breaking change.

We should do a precursor PR to add Obj-C unit tests of this method, with a comment explaining that they are testing a public API and must continue to work unchanged, and then preserve the Obj-C tests even in the Swift conversion to be sure we don't accidentally change the signature.

Reply:
> We should do a precursor PR to add Obj-C unit tests of this method, with a comment explaining that they are testing a public API and must continue to work unchanged, and then preserve the Obj-C tests even in the Swift conversion to be sure we don't accidentally change the signature.

https://github.com/flutter/packages/pull/4925